### PR TITLE
fix: achievement_reward_locale bug

### PR DESCRIPTION
**Pull request by AsunaFrostwyrm**
## CHANGES PROPOSED:
-  Added the locale based build of the MailTemplate in the AchievementMgr.cpp

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2810

## TESTS PERFORMED:
- Built successfully
- Tested ingame, received german mails after completing achievements while using a german client

## HOW TO TEST THE CHANGES:

1. Download or use an other languaged client (not english)
2. Complete an achievement where u receive a mail, like reaching LvL 80
3. Read the mail you'll receive

## KNOWN ISSUES AND TODO LIST:

- None

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
thank you for that PR!

**Comment by FrancescoBorzi** [comment]
@AsunaFrostwyrm good jo!

@pklloveyou @solidmaxtor can you please have a look at this?

**Comment by Si1ker** [comment]
@AsunaFrostwyrm It copiled for me in Win 10 Pro (esES client)

I've used the following change:
```c++
        if (!reward->mailTemplate)
        {
	    std::string subject = reward->subject;
	    std::string text = reward->text;

	    LocaleConstant localeConstant = GetPlayer()->GetSession()->GetSessionDbLocaleIndex();
	    if (localeConstant != LOCALE_enUS)
	    {
	        if(AchievementRewardLocale const* loc = sAchievementMgr->GetAchievementRewardLocale(achievement))
		{
		    ObjectMgr::GetLocaleString(loc->Subject, localeConstant, subject);
		    ObjectMgr::GetLocaleString(loc->Text, localeConstant, text);
		}
	    }

            draft = MailDraft(subject, text);
        }

        SQLTransaction trans = CharacterDatabase.BeginTransaction();
```

It didn't work for me, I'm still getting the english mail.
2020-03-28 15_38_36-World of Warcraft

**Comment by Nefertumm** [comment]
Make sure that the locale data exists in db, if not then the pr doesn't work and need some kind of improvement.

**Comment by AsunaFrostwyrm** [comment]
Exactly, in case there is no row in the achievement_reward_locale table with the fitting ID it will send the common english mail.

What I inserted in my achievement_reward_locale is this row:

```SQL
INSERT INTO `achievement_reward_locale` VALUES (13, 'deDE', 'Glückwunsch', 'Viele Glückwunsche bezüglich eurer Überzeugung, auch die 80. Saison der Abenteurer zu erreichen. Ihr seit zweifellos dem Ziel ergben, Azeroth vom Bösen zu befreien.$B$BObwohl der Weg hierher keine Kleinigkeit war, steht der wahre Kampf erst noch bevor.$B$BKämpft weiter!$B$BRhonin');
```

Then, when I reached level 80 I got this mail:

image


Of course the "locale" should be setted to your specific locale-value, in your case it's esES I think?

**Comment by Si1ker** [comment]
Yes, a good thing to keep in mind. 🐱 
After this has been merged I will include the other clients.

It is working for me in esES client with the folowing:
```sql
INSERT INTO `achievement_reward_locale` VALUES (13, 'esES', 'Nivel 80', 'Enhorabuena por tu convicción de alcanzar la 80ª temporada de aventuras. Sin duda, estás dedicado a la causa de librar a Azeroth de los males que nos han acosado.$B$BY aunque el viaje hasta el momento no ha sido una hazaña menor, la verdadera batalla está por delante.$B$B¡Lucha!$B$BRhonin');
```
2020-03-28 17_20_24-World of Warcraft

Good job Malinka. 🚀

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 1]
```diff
@@ -2219,8 +2219,23 @@ void AchievementMgr::CompletedAchievement(AchievementEntry const* achievement)
         MailDraft draft(reward->mailTemplate);
 
         if (!reward->mailTemplate)
-            draft = MailDraft(reward->subject, reward->text);
-
+		{
```
Comment (Nefertumm) [inline src/server/game/Achievements/AchievementMgr.cpp]: Fix identation before merge please.

**Comment by Nefertumm** [comment]
We use 4 spaces as indentation, watch out for that. 
Thank you very much for your contribution :)

**Comment by BarbzYHOOL** [comment]
renamed your commit to `fix(CORE/locale): achievement_reward_locale should now work (#2811)` and merged it, thank you very much @AsunaFrostwyrm 

are you on our discord? (i think so, but which name?)

**Comment by BarbzYHOOL** [comment]
also @Si1ker you said you wanted to add smthg else after that, just reminding you

**Comment by Si1ker** [comment]
@BarbzYHOOL @AsunaFrostwyrm  Yeah, I've have bookmarked this PR, once it gets merged I will add the email in different languages because to test this one, you have to create it manually.
I'll wait for it 🐱

**Comment by AsunaFrostwyrm** [comment]
@BarbzYHOOL Yeah, I joined the discord at the day I created this PR.

Name there is: Malinka - AsunaFrostwyrm, I already got the Contributor LvL 1 Rank.

**Comment by BarbzYHOOL** [comment]
done?

**Pull request merged**